### PR TITLE
Feature/cor 378 authmodule   notification if pub

### PR DESCRIPTION
**Pull request by zextrasGiovanniDefacci**

### CODE REVIEW

File: src/java/org/openzal/zal/ProvisioningImp.java [hunk 1]
```diff
@@ -40,66 +29,98 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
 import org.dom4j.QName;
-import org.openzal.zal.exceptions.*;
```
Comment (mzanellazextras) [inline src/java/org/openzal/zal/ProvisioningImp.java:62]: I am not sure that removing the imports with "*" make all work with any version. Have you tried to compile it for each version?

File: src/java/org/openzal/zal/ProvisioningImp.java [hunk 2]
```diff
@@ -40,66 +29,98 @@
...
-import com.zimbra.cs.mailbox.Contact;
-
-import javax.annotation.Nullable;
-import org.openzal.zal.log.ZimbraLog;
-import org.openzal.zal.provisioning.DirectQueryFilterBuilder;
-
 public class ProvisioningImp implements Provisioning
 {
   /* $if ZimbraVersion >= 8.7.0 $ */
-  public static String A_zimbraMaxAppSpecificPasswords                        = com.zimbra.cs.account.Provisioning.A_zimbraMaxAppSpecificPasswords;
```
Comment (mzanellazextras) [inline src/java/org/openzal/zal/ProvisioningImp.java:95]: Avoid useless whitespaces changes

File: src/java/org/openzal/zal/ProvisioningImp.java (hunk 1 above)
Comment (zextrasGiovanniDefacci) [reply src/java/org/openzal/zal/ProvisioningImp.java]: Thank you for noticing it, I restored the initial import and formatting.

File: src/java/org/openzal/zal/ProvisioningImp.java (hunk 2 above)
Comment (zextrasGiovanniDefacci) [reply src/java/org/openzal/zal/ProvisioningImp.java]: Thank you for noticing it, I restored the initial import and formatting.

**Pull request closed without merging**